### PR TITLE
fix user-cluster Prometheus not scraping kube-state-metrics

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
During refactorings we removed the Service for kube-state-metrics. As the service was the only element that contained Prometheus scraping annotations, we have since then not scraped it anymore.

This PR restores the annotation directly on the Deployment's spec. It also defines a new alert so that we notice the missing scrape target earlier.

**Does this PR introduce a user-facing change?**:
```release-note
fixed kube-state-metrics in user-clusters not being scraped
```

### CODE REVIEW

File: api/pkg/resources/kubestatemetrics/deployment.go [hunk 1]
```diff
@@ -53,6 +53,11 @@ func DeploymentCreator(data *resources.TemplateData) reconciling.NamedDeployment
 
 			dep.Spec.Template.ObjectMeta = metav1.ObjectMeta{
 				Labels: podLabels,
+				Annotations: map[string]string{
+					// do not specify a port so that Prometheus automatically
```
Comment (alvaroaleman) [inline api/pkg/resources/kubestatemetrics/deployment.go:57]: @xrstf Prometheus can get multiple targets out of a single annotation? Is that new?

Comment (mrIncompetent) [reply api/pkg/resources/kubestatemetrics/deployment.go:57]: Prometheus creates a target for every port:
https://github.com/prometheus/prometheus/blob/master/discovery/kubernetes/pod.go#L229

Comment (xrstf) [reply api/pkg/resources/kubestatemetrics/deployment.go:57]: I was surprised as well, but that how it works after adding the annotation :-)

**Pull request merged**